addons/web/static/src/js/view_list_editable.js

```diff
@@ -257,7 +257,15 @@ openerp.web.list_editable = function (instance) {
                         if (!focus_field){
                             focus_field = _.find(self.editor.form.fields_order, function(field){ return fields[field] && fields[field].$el.is(':visible:has(input)'); });
                         }
-                        if (focus_field  && fields[focus_field]) fields[focus_field].$el.find('input').select();
+                        if (focus_field  && fields[focus_field]){
+                            var input = fields[focus_field].$el.find('input');
+                            if(input[0] && input[0].type === 'text' ||
+                               input[0].type === 'textarea'){
```

Comment:
Looks like input[0].type === 'textarea' is still going to be evaluated even if input[0] is not defined, due to operator precedence:
```
$ false && false || true
true
```
Either this test is unnecessary, or you will want to use some parentheses.